src/v2/view-builder/views/google-authenticator/EnrollAuthenticatorGoogleAuthenticatorView.js

```diff
@@ -0,0 +1,99 @@
+import { loc, createButton } from 'okta';
+import BaseForm from '../../internals/BaseForm';
+import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
+import AuthenticatorEnrollFooter from '../../components/AuthenticatorEnrollFooter';
+import EnrollGoogleAuthenticatorBarcodeView from './EnrollGoogleAuthenticatorBarcodeView';
+import EnrollAuthenticatorMannuallySetupView from './EnrollAuthenticatorMannuallySetupView';
+
+const Body = BaseForm.extend({
+  title () {
+    return loc('oie.enroll.google_authenticator.setup.title', 'login');
+  },
+
+  subtitle () {
+    return loc('oie.enroll.google_authenticator.enterCode.title', 'login');
+  },
+
+  save () {
+    return loc('mfa.challenge.verify', 'login');
+  },
+
+  className: 'oie-enroll-google-authenticator',
+
+  modelEvents: {
+    'triggerBarcodeView': '_triggerBarcodeView'
+  },
+
+  _triggerBarcodeView () {
+    this.$('.oie-enroll-google-authenticator-barcode').hide();
+    this._showManuallySetupView();
+  },
+
+  _showVerifyCodePage () {
+    this.$('.oie-enroll-google-authenticator-barcode').hide();
+    this.$('.google-authenticator-next').hide();
+    this._hideManuallySetupView();
+    this.$('.o-form-label-top').show();
+    this.$('.o-form-button-bar').show();
+  },
+
+  _hideVerifyCodePage () {
+    this.$('.o-form-label-top').hide();
+    this.$('.o-form-button-bar').hide();
+    this.$('.okta-form-subtitle').hide();
+  },
+
+  _showManuallySetupView () {
+    this.$('.shared-secret').show();
+    this.$('.oie-enroll-google-authenticator-manual-setup').show();
+  },
+
+  _hideManuallySetupView () {
+    this.$('.shared-secret').hide();
+    this.$('.oie-enroll-google-authenticator-manual-setup').hide();
+  },
+
+  render () {
+    BaseForm.prototype.render.apply(this, arguments);
+    this.$('.oie-enroll-google-authenticator-manual-setup').hide();
+    this.$('.shared-secret').hide();
+    this._hideVerifyCodePage();
+  },
+
+  getUISchema () {
+    const schema = BaseForm.prototype.getUISchema.apply(this, arguments);
+
+    const nextButton = createButton({
+      className: 'google-authenticator-next button-primary default-custom-button',
+      title: loc('oform.next', 'login'),
+      click: () => {
```

Comment:
try this to see if it works click : this._showVerifyCodePage

Reply:
Unfortunately we cannot, the `this` changed

Reply:
See if you can use `createCustomButtons` from FromInputFactory.. Also I would suggest to add the function to create button in FormInputFactory.js

Reply:
Looks we cannot when we use `showWhen` we need to customize the style.

Reply:
you can update `createCustomButtons` function to accept "style" as well..
```
attribute: customButton.attribute
```

Reply:
No. I don't think we can do this.

`FromInputFactory` takes the setting from options and generates a settings object that passes to some child views to add the button, even in the child views they explicitly call the `add` ([line](https://github.com/okta/okta-signin-widget/blob/master/src/v2/view-builder/views/signin/CustomButtons.js#L14)). We did different here that explicitly add a button, `createCustomButtons` doesn't apply. We can later add a util function to add some generic button, but since there is a lot of usage of `createButton`, we'd better refactor later for all of these usages.

For now, I've added a private util function on the view to make it easier to read. 
I think it is over-engineering in this pr for creating a general util function accomodating two buttons in the same class.

Reply:
responded over slack..